elfpy/utils/math/fixed_point.py

```diff
@@ -48,24 +55,26 @@ def _coerce_other(self, other):
             Right now we do not support operating against int and float because those are logically confusing
         """
         if isinstance(other, FixedPoint):
+            if other.special_value is not None:
+                return FixedPoint(other.special_value)
             return other
         if isinstance(other, (int, float)):  # currently don't allow floats & ints
             raise TypeError(f"unsupported operand type(s): {type(other)}")
         return NotImplemented
 
-    def __int__(self) -> int:
-        """Cast to int"""
-        return self.int_value
-
-    def __float__(self) -> float:
-        """Cast to float"""
-        return float(self.int_value) / 10**self.decimal_places
-
     def __add__(self, other: int | FixedPoint) -> FixedPoint:
         """Enables '+' syntax"""
         other = self._coerce_other(other)
         if other is NotImplemented:
             return NotImplemented
+        if self.is_nan() or other.is_nan():
+            return FixedPoint("nan")
+        if self.is_inf():
+            if other.is_inf() and self.sign() != other.sign():
```

Comment:
inf * -inf = nan? TIL

Reply:
Not quite! It's easiest to see the rules if you look at the test. In the `test_fixed_point_nonfinite.py` file you can see: `assert self.NEG_INF * self.INF == self.NEG_INF`